### PR TITLE
LPS-126596 Simpler solution for OAuth2 Trusted Applications

### PR DESCRIPTION
**Pull request by martamedio**
Hey @stian-sigvartsen 

During the analysis of the "_Remember by Device_" issue, @csierra has seen that [this method could be overwritten](https://github.com/liferay-appsec/liferay-portal/commit/ac1db8657c1b79431d5ade0732778f111b03c210#diff-30016c557f4dede731fe82a8a20dfae624b4f7e5a628ca0a0b433d2d2cc75181R77), so this makes the solution for checking trusted applications much simpler and saves us the steps related to modify the redirect at _OAuthAuthorizationDataMessageBodyWriter_.

Let me know what you think!

### CODE REVIEW

File: modules/apps/oauth2-provider/oauth2-provider-api/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -4,6 +4,7 @@ CLIENT_CREDENTIALS=Client Credentials
 REFRESH_TOKEN=Refresh Token
```
Comment (liferay-appsec) [inline modules/apps/oauth2-provider/oauth2-provider-api/src/main/resources/content/Language.properties:4]: Check that added properties are either titles or sentences

File: modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/admin/edit_application_left_column.jsp [hunk 2]
```diff
@@ -118,6 +118,9 @@ OAuth2Application oAuth2Application = oAuth2AdminPortletDisplayContext.getOAuth2
 								<c:when test="<%= grantType.equals(GrantType.CLIENT_CREDENTIALS) %>">
 									<aui:input checked="<%= checked %>" data="<%= data %>" helpMessage="the-client-will-impersonate-the-selected-client-credential-user-but-will-be-restricted-to-the-selected-scopes" label="<%= grantType.name() %>" name="<%= clientCredentialsCheckboxName %>" onchange='<%= liferayPortletResponse.getNamespace() + "updateClientCredentialsSection();" %>' type="checkbox" />
 								</c:when>
+								<c:when test="<%= grantType.equals(GrantType.AUTHORIZATION_CODE) || grantType.equals(GrantType.AUTHORIZATION_CODE_PKCE) %>">
+									<aui:input checked="<%= checked %>" data="<%= data %>" label="<%= grantType.name() %>" name="<%= name %>" onchange='<%= liferayPortletResponse.getNamespace() + "updateTrustedApplicationSection();" %>' type="checkbox" />
```
Comment (liferay-appsec) [inline modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/admin/edit_application_left_column.jsp:122]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/admin/edit_application_left_column.jsp [hunk 3]
```diff
@@ -221,6 +224,19 @@ OAuth2Application oAuth2Application = oAuth2AdminPortletDisplayContext.getOAuth2
...
 		</c:if>
+
+		<c:if test="<%= oAuth2AdminPortletDisplayContext.hasAddTrustedApplicationPermission() %>">
+			<clay:col
+				id='<%= liferayPortletResponse.getNamespace() + "trustedApplicationSection" %>'
+				lg="12"
+			>
+				<h3 class="sheet-subtitle"><liferay-ui:message key="trusted-application" /></h3>
+
+				<aui:field-wrapper>
```
Comment (liferay-appsec) [inline modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/admin/edit_application_left_column.jsp:235]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/content/Language.properties [hunk 4]
```diff
@@ -97,6 +97,8 @@ this-application-wants-the-following-permissions=This application wants the foll
 this-operation-cannot-be-performed-because-you-cannot-impersonate-x=This operation cannot be performed because you cannot impersonate user {0}.
```
Comment (liferay-appsec) [inline modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/content/Language.properties:97]: Check that added properties are either titles or sentences

**Review by liferay-appsec — COMMENTED** [review]
This is an automatic review suggestion

**Comment by martamedio** [comment]
ci:test:sf

**Comment by martamedio** [comment]
ci:test:relevant

**Comment by stian-sigvartsen** [comment]
@martamedio That's clearly a much better approach! Nice!

**Comment by csierra** [comment]
it LGTM too... can we only squash https://github.com/liferay-appsec/liferay-portal/pull/279/commits/22aa66f189c3a818e9209be5b6c9e7a77682308f on top of https://github.com/liferay-appsec/liferay-portal/pull/279/commits/b2f8429b6cd3c40873cd59d2ef5af549676a0abb ? If we don't a lot of classes appear to have changed until you look inside and then there are no changes...

**Comment by csierra** [comment]
ci:forward

**Comment by csierra** [comment]
ci:forward

**Comment by csierra** [comment]
ci:stop

**Comment by csierra** [comment]
ci:forward

**Pull request closed without merging**